internal/providers/terraform/azure/app_service_environment.go

```diff
@@ -0,0 +1,101 @@
+package azure
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/infracost/infracost/internal/schema"
+
+	"github.com/shopspring/decimal"
+)
+
+func GetAzureRMAppIsolatedServicePlanRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "azurerm_app_service_environment",
+		RFunc: NewAzureRMAppIsolatedServicePlan,
+		ReferenceAttributes: []string{
+			"resource_group_name",
+		},
+		Notes: []string{
+			"Costs associated with running an app service plan in Azure",
+		},
+	}
+}
+
+func NewAzureRMAppIsolatedServicePlan(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: AppIsolatedServicePlanCostComponent(d, u),
+	}
+}
+func AppIsolatedServicePlanCostComponent(d *schema.ResourceData, u *schema.UsageData) []*schema.CostComponent {
+
+	tier := d.Get("pricing_tier").String()
```

Comment:
`pricing_tier` [is optional](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/app_service_environment#pricing_tier) in terraform, so this should default to `I1` according to the terraform docs and a check should be done like `if d.Get("pricing_tier").Type != gjson.Null` to set its value